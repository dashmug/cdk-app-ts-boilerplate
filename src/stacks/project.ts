import type { StackProps } from "aws-cdk-lib";
import {
  AttributeType,
  Table,
  TableEncryption,
} from "aws-cdk-lib/aws-dynamodb";
import type { Construct } from "constructs";
import { BaseStack } from "src/lib/base-stack";

/**
 * The stack that defines the resources for the project.
 */
export class MyProjectStack extends BaseStack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    // Change this to suit your project.
    new Table(this, "Table", {
      partitionKey: { name: "pk", type: AttributeType.STRING },
      sortKey: { name: "sk", type: AttributeType.STRING },
      pointInTimeRecovery: true,
      encryption: TableEncryption.CUSTOMER_MANAGED,
    });
  }
}
