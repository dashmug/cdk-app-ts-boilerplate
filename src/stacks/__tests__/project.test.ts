import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { MyProjectStack } from "src/stacks/project";

describe("MyProjectStack", () => {
  const app = new App();
  const stack = new MyProjectStack(app, "MyTestStack");
  const template = Template.fromStack(stack);

  it("matches the default snapshot", () => {
    expect(template).toMatchSnapshot();
  });
});
