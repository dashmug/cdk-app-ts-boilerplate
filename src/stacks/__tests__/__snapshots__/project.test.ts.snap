// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProjectStack matches the default snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "TableCD117FA1": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "pk",
            "AttributeType": "S",
          },
          {
            "AttributeName": "sk",
            "AttributeType": "S",
          },
        ],
        "KeySchema": [
          {
            "AttributeName": "pk",
            "KeyType": "HASH",
          },
          {
            "AttributeName": "sk",
            "KeyType": "RANGE",
          },
        ],
        "PointInTimeRecoverySpecification": {
          "PointInTimeRecoveryEnabled": true,
        },
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "SSESpecification": {
          "KMSMasterKeyId": {
            "Fn::GetAtt": [
              "TableKey25666F95",
              "Arn",
            ],
          },
          "SSEEnabled": true,
          "SSEType": "KMS",
        },
        "Tags": [
          {
            "Key": "myorg:application:description",
            "Value": "Opinionated CDK Typescript Boilerplate",
          },
          {
            "Key": "myorg:application:environment",
            "Value": "prod",
          },
          {
            "Key": "myorg:application:name",
            "Value": "cdk-app-ts-boilerplate",
          },
          {
            "Key": "myorg:application:repository",
            "Value": "https://github.com/dashmug/cdk-app-ts-boilerplate",
          },
          {
            "Key": "myorg:project:customer",
            "Value": "myclient",
          },
          {
            "Key": "myorg:project:name",
            "Value": "myproject",
          },
          {
            "Key": "myorg:project:owner",
            "Value": "myteam",
          },
          {
            "Key": "myorg:stack:name",
            "Value": "MyTestStack",
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "TableKey25666F95": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "Description": "Customer-managed key auto-created for encrypting DynamoDB table at MyTestStack/Table",
        "EnableKeyRotation": true,
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "myorg:application:description",
            "Value": "Opinionated CDK Typescript Boilerplate",
          },
          {
            "Key": "myorg:application:environment",
            "Value": "prod",
          },
          {
            "Key": "myorg:application:name",
            "Value": "cdk-app-ts-boilerplate",
          },
          {
            "Key": "myorg:application:repository",
            "Value": "https://github.com/dashmug/cdk-app-ts-boilerplate",
          },
          {
            "Key": "myorg:project:customer",
            "Value": "myclient",
          },
          {
            "Key": "myorg:project:name",
            "Value": "myproject",
          },
          {
            "Key": "myorg:project:owner",
            "Value": "myteam",
          },
          {
            "Key": "myorg:stack:name",
            "Value": "MyTestStack",
          },
        ],
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
